kernel/rtlil.cc

```diff
@@ -365,6 +365,20 @@ std::vector<RTLIL::State> RTLIL::Const::to_bits() const
 	return b;
 }
 
+std::string RTLIL::Const::pretty_fmt() const {
+	if (get_if_str())
```

Comment:
This leaks the backing

Reply:
Yes, it's used for pretty printing, which isn't supposed to be some kind of correctness-dependent process, but as a compact string representation readable by a human

Reply:
There's no guarantee someone will not be relying on the output not changing if the underlying value hasn't changed. Case in point: you have used it this way for hashing in opt_merge.cc

Reply:
Good point, I didn't consider that the same string packed and unpacked then has to be equal in hashing input as well